### ISSUE TITLE
[1.9.5+] \ref used in external doc fails to find some functions

### ISSUE THREAD

**Issue by jzern** [doxygen/doxygen#10895]
**Describe the bug**
doc/dev_guide/av1_encoder.dox:590: warning: unable to resolve reference to 'av1_calc_iframe_target_size_one_pass_vbr()' for \ref command
doc/dev_guide/av1_encoder.dox:591: warning: unable to resolve reference to 'av1_calc_pframe_target_size_one_pass_vbr()' for \ref command
doc/dev_guide/av1_encoder.dox:704: warning: unable to resolve reference to 'av1_rc_regulate_q()' for \ref command

**Screenshots**
N/A

**To Reproduce**
I don't have a simple case to reproduce the issue. This shows up in the libaom project.

```
$ git clone -b karat --depth=1 https://aomedia.googlesource.com/aom
$ cmake -S aom -B aom-build -DENABLE_TESTS=0 -DENABLE_EXAMPLES=0
$ (cd aom-build; doxygen doxyfile)
```

**Note** doxygen must be in your PATH for aom-build/doxyfile to be created by cmake. `ninja -C aom-build` will also run doxygen.

**Expected behavior**
Links are resolved.

**Version**
\>= 1.9.5 (tested with 2dccfe1ae9fc49bb5ccc2a7baf5c5ccb4fec816b). Since Release_1_9_4-50-gb290399fa.

```
b290399fab92df45376103d7c094f053f24495fa is the first bad commit
commit b290399fab92df45376103d7c094f053f24495fa (HEAD)
Author: Dimitri van Heesch <doxygen@gmail.com>
Date:   Tue May 24 08:12:07 2022 +0200

    Refactoring: reimplement getDefs using symbol resolver

 src/classdef.cpp       |  44 +++++---
 src/classdef.h         |   6 +-
 src/code.l             |   2 +-
 src/definition.cpp     |   1 +
 src/doxygen.cpp        |  40 +++++---
 src/doxygen.h          |   9 +-
 src/symbolresolver.cpp | 568 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-----------
 src/symbolresolver.h   |  11 ++
 src/util.cpp           | 173 +++++++++++++++++++++++++++----
 src/vhdldocgen.cpp     |  12 ++-
 10 files changed, 741 insertions(+), 125 deletions(-)
```

**Stack trace**
N/A

**Additional context**
The references are in [doc/dev_guide/av1_encoder.dox](https://aomedia.googlesource.com/aom/+/refs/heads/karat/doc/dev_guide/av1_encoder.dox). They refer to functions in [av1/encoder/ratectrl.h](https://aomedia.googlesource.com/aom/+/refs/heads/karat/av1/encoder/ratectrl.h). Note other functions from this header resolve without issue. Adding a suffix to the function names, like '2', appears to work as well.

As a workaround using a full reference to the functions works, e.g.,  `\ref av1_calc_iframe_target_size_one_pass_vbr(const struct AV1_COMP *const cpi)`.

**Comment by jzern** [comment]
cc: @wantehchang

**Comment by albert-github** [comment]
I see you bisected the problem and concluded the problem was introduced with commit ttps://github.com/doxygen/doxygen/commit/2dccfe1ae9fc49bb5ccc2a7baf5c5ccb4fec816b 
- Did you also try the current doxygen version (1.11.0) or the current doxygen master?

You refer  to an external and potential unstable source as reproducer, so:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by albert-github** [comment]
I think the problem is due to the `typedef` that is defined in 

**encoder.h**
```
typedef struct AV1_COMP {
...
} AV1_COMP;
```
and the declarations and definition in

**ratectrl.h**
```
/*!\brief Calculates how many bits to use for an i frame in one pass vbr
 *
 * \ingroup rate_control
 * \callgraph
 * \callergraph
 *
 * \param[in]       cpi  Top level encoder structure
 *
 * \return      Returns the target number of bits for this frame.
 */
int av1_calc_iframe_target_size_one_pass_vbr(const struct AV1_COMP *const cpi);
```

**ratectrl.c**
```
int av1_calc_iframe_target_size_one_pass_vbr(const AV1_COMP *const cpi) {
...
````

with the use of the `typedef` these 2 signatures are the same but it looks like doxygen doesn't apply the expansion resulting in the observed behavior.

**Comment by jzern** [comment]
> I see you bisected the problem and concluded the problem was introduced with commit ttps://github.com/doxygen/doxygen/commit/2dccfe1ae9fc49bb5ccc2a7baf5c5ccb4fec816b
> 
> * Did you also try the current doxygen version (1.11.0) or the current doxygen master?
> 

Yes, I mentioned that in the report. I tested with 2dccfe1ae9fc49bb5ccc2a7baf5c5ccb4fec816b; the current master at the time of report.

> You refer to an external and potential unstable source as reproducer, so:
> 
> * Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

Sorry, I didn't try to minimize this. libaom's repo should be quite stable, but I've attached a snapshot of the 3.9.0 release which is based on the branch I linked in the report.
[libaom-3.9.0.tar.gz](https://github.com/doxygen/doxygen/files/15444295/libaom-3.9.0.tar.gz)
[libaom-3.9.0.tar.gz.asc.gz](https://github.com/doxygen/doxygen/files/15444297/libaom-3.9.0.tar.gz.asc.gz)

**Comment by jzern** [comment]
> with the use of the typedef these 2 signatures are the same but it looks like doxygen doesn't apply the expansion resulting in the observed behavior.

Thanks for having a look. I noticed other functions resolve from that source and header pair like:

```
void av1_rc_update_rate_correction_factors(struct AV1_COMP *cpi,
                                           int is_encode_stage, int width,
                                           int height);
void av1_rc_update_rate_correction_factors(AV1_COMP *cpi, int is_encode_stage,
                                           int width, int height) {
```

Also, slightly changing the name to the functions, by adding a '2' suffix for instance, seemed to work.

**Comment by albert-github** [comment]
I just ran the given [libaom-3.9.0.tar.gz](https://github.com/doxygen/doxygen/files/15444295/libaom-3.9.0.tar.gz) with the instructions from https://github.com/doxygen/doxygen/issues/10895#issue-2316321892
```
$ cmake -S libaom-3.9.0 -B aom-build -DENABLE_TESTS=0 -DENABLE_EXAMPLES=0
$ (cd aom-build; doxygen doxyfile)
```
Some links are now corrected but also some new problems occurred between the doxygen 1.11.0 version and the current master (1.12.0 (a6988ea47d4feafb63e18f5bcf5bbbcc656e7edc), e.g no link is now present at e.g. html/example_aomenc.html#l00070

Interesting is also that now theer is no code coloring of the word `if` on the lines 75 and 77 from the same file

**Comment by jzern** [comment]
> [...] e.g no link is now present at e.g. html/example_aomenc.html#l00070

Try removing `-DENABLE_EXAMPLES=0` from the cmake line. I was just giving a minimal configuration to reproduce the original issue, but this may affect the doxygen output.
